src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportJobWorker.cs

```diff
@@ -69,16 +72,31 @@ public async Task ExecuteAsync(CancellationToken cancellationToken)
                         }
                     }
 
-                    await Task.Delay(_exportJobConfiguration.JobPollingFrequency, cancellationToken);
-                }
-                catch (OperationCanceledException) when (cancellationToken.IsCancellationRequested)
-                {
-                    // Cancel requested.
+                    // We successfully completed an attempt to acquire export jobs. Let us reset the polling frequency in case it has changed.
+                    delayBeforeNextPoll = _exportJobConfiguration.JobPollingFrequency;
                 }
                 catch (Exception ex)
                 {
                     // The job failed.
                     _logger.LogError(ex, "Unhandled exception in the worker.");
+
+                    // Since acquiring jobs failed let us introduce a delay before we retry. We don't want to increase the delay between polls to more than an hour.
+                    delayBeforeNextPoll *= 2;
+                    if (delayBeforeNextPoll.TotalSeconds > MaximumDelayInSeconds)
+                    {
+                        delayBeforeNextPoll = TimeSpan.FromSeconds(MaximumDelayInSeconds);
+                    }
+                }
+
+                try
+                {
+                    await Task.Delay(delayBeforeNextPoll, cancellationToken);
+                }
+                catch (OperationCanceledException) when (cancellationToken.IsCancellationRequested)
+                {
+                    // Cancellation requested
+                    _logger.LogInformation("ExportJobWorker: Cancellation requested.");
```

Comment:
nit: "while (!cancellationToken.IsCancellationRequested)" exit path doesn't have this log entry. Should we remove (or add info in message of) the log here to avoid confusion while analyzing logs?

Reply:
Since both scenarios are caused by the same underlying cancellation (`Task.Delay()` throws the `TaskCanceledException` when `cancellationToken.IsCancellationRequested == true`), I moved the log outside the while loop. Now the logging will be uniform regardless of where we exit the loop.